src/Config/ConfigMerger.php

```diff
@@ -42,14 +42,24 @@ public function merge(Config $config, Config $override = null): Config
     private function mergeConfigEnvironments(Config $config, Config $override): array
     {
         $environments = [];
-        foreach ($override->getEnvironments() as $name => $overrideEnv) {
-            $originalConfigEnv = $config->getEnvironments()[$name];
+
+        $foundEnvironments = array_keys(array_merge($config->getEnvironments(), $override->getEnvironments()));
+
+        foreach ($foundEnvironments as $name) {
+            if (!isset($override->getEnvironments()[$name])) {
+                $environments[$name] = $config->getEnvironments();
+
+                continue;
+            }
+
+            $overrideEnv = $override->getEnvironments()[$name];
+            $originalConfigEnv = $config->getEnvironments()[$name] ?? new ConfigEnvironment();
```

Comment:
I am not entirely sure if an empty object will work here. I will merge this onto the 1.5 branch and test it from there.